packages/terra-cli/index.js

```diff
@@ -0,0 +1,119 @@
+        // the recommendCommands() message is too terse
+        if (/Did you mean/.test(actual.message)) {
+          // eslint-disable-next-line no-console
+          logger.error(`Unknown command "${cli.parsed.argv._[0]}"`);
+        }
+
+        // eslint-disable-next-line no-console
+        logger.error(actual.message);
+      }
+
+      // exit non-zero so the CLI can be usefully chained
+      cli.exit(actual.code > 0 ? actual.code : 1, actual);
+    })
+    .alias('h', 'help')
+    .alias('v', 'version')
+    .wrap(cli.terminalWidth());
+};
+
+/**
+ * Takes a directory path and finds all immediate sub directories
+ * @param {string} directoryPath - the directory path for which to find sub directories
+ * @returns {Array} an array of the sub directories appended to the passed in directory path
+ */
+const subDirectoriesOfDirectory = async (directoryPath) => {
+  // Gets contents of current directory
+  const currentDirectoryEnries = await fs.readdir(directoryPath, { withFileTypes: true });
+  // Filters down to just the directories and then appends the directory names to the initial directory path
+  return currentDirectoryEnries.filter(directoryEntry => directoryEntry.isDirectory())
+    .map(directoryEntry => path.join(directoryPath, directoryEntry.name));
+};
+
+/**
+ * Takes an array of paths, appends either src or lib depending on preferSrc, appends terra-cli, and then returns an array of the the resulting directory paths that exist
+ * @param {Array} paths - the array of paths to act upon
+ * @param {boolean} preferSrc - indicates whether or not src should be used instead of lib. This should be used in cases where the package being searched is local
+ * @returns {Array} an array of paths resolved out to the terra-cli directory
+ */
+const terraClIDirectoriesFromPaths = async (paths, preferSrc) => {
+  const cliDirectories = paths.map(dependencyPath => path.join(dependencyPath, preferSrc ? 'src' : 'lib', 'terra-cli'));
+  const pathExistsForDirectories = await Promise.all(cliDirectories.map((directory) => fs.pathExists(directory)));
+  return cliDirectories.filter((_, index) => pathExistsForDirectories[index]);
+};
+
+/**
+ * Searches for terra-cli command paths from the current directory following these rules
+ *
+ * 1. Dependencies in the top node_modules directory that are in the dependencies allow list
+ * 2. Monorepo package directories if the current project is terra-toolkit and the directory is in the dependency allow list
+ * 3. The current directory if its in the dependencies allow list
+ * @returns {Array} an array of all qualifying terra-cli command paths
+ */
+const loadTerraCommandPaths = async () => {
+  const { name } = fs.readJSONSync(path.join(process.cwd(), 'package.json'));
+  // Search the top level node_modules for dependencies in the allow list
+  const firstLevelDependencyPaths = DEPENDENCY_ALLOW_LIST.map(dependency => path.join(process.cwd(), 'node_modules', dependency));
```

Comment:
do you have to take the @cerner scoping into account here? Both for the package name well as the path?

Reply:
I have no updated this to fix this.  I use `@cerner` but then drop it when looking inside of the terra toolkit monorepo

Reply:
That covers the name from the package, but in node modules `@cerner` scoped packages will be under the `@cerner` directory. You'll have to split the name off the dependency.... or cheat and only search the `@cerner` directory.

```suggestion
  const firstLevelDependencyPaths = DEPENDENCY_ALLOW_LIST.map(dependency => path.join(process.cwd(), 'node_modules', '@cerner', dependency));
```

Reply:
lol, never mind, this is already covered